skimage/draw/draw3d.py

```diff
@@ -63,6 +64,165 @@ def ellipsoid(a, b, c, spacing=(1., 1., 1.), levelset=False):
+
+    Parameters
+    ----------
+    center : (3,) array-like of float
+        Center coordinate of ellipsoid. The values will
+        be divided by ``spacing`` inside the function.
+    axis_lengths : (3,) array-like of float
+        Axis lengths of ellipsoid. The order is (z, y, x). The values will be
+        divided by ``spacing`` inside the function.
+    shape : tuple of int, length 3, optional
+        Image shape which is used to determine the maximum extent of output
+        pixel coordinates. This is useful for ellipsoids which exceed the
+        image size. By default the full extent of the ellipsoid is used.
+    angles : iterable of float, length 3, optional (default None)
+        Rotation angles to rotate the ellipsoid. The axes about which to
+        produce the rotations are specified by ``axes``. No rotation will be
+        applied if ``angles`` is ``None``.
+    axes : iterable of int, length 3, optional (default None)
+        Axes about which to produce the rotations.
+        Defaults to ``[0, 1, 2]``.
+    intrinsic : bool, optional
+        Specify if the rotations are intrinsic (``True``) or extrinsic
+        (``False``).
+
+        - *intrinsic*: rotations occur about the rotating coordinate system
+        - *extrinsic*: rotations occur about the original coordinate system
+    spacing : (3,) array-like of float, optional
+        Spacing in each spatial dimension.  Defaults to 1.
+
+    Returns
+    -------
+    dd, rr, cc : ndarray of int
+        Voxel coordinates of ellipsoid.
+        May be used to index into an array, e.g.
+        ``img[dd, rr, cc] = 1``
+
+    Notes
+    ------
+    ``angles`` are defined as the Euler angles based on ``axes`` and
+    ``intrinsic`` parameters. The following example shows the ``0-2-0``
+    (``z-x-z``) intrinsic rotations.
+
+    .. math::
+
+        R(\phi) = \begin{bmatrix}
+                      1 &        0 &         0 \\
+                      0 & \cos\phi & -\sin\phi \\
+                      0 & \sin\phi &  \cos\phi \\
+                  \end{bmatrix}
+
+        R(\theta) = \begin{bmatrix}
+                        \cos\theta & -\sin\theta & 0 \\
+                        \sin\theta &  \cos\theta & 0 \\
+                                 0 &           0 & 1 \\
+                    \end{bmatrix}
+
+        R(\psi) = \begin{bmatrix}
+                      1 &        0 &         0 \\
+                      0 & \cos\psi & -\sin\psi \\
+                      0 & \sin\psi &  \cos\psi \\
+                           \end{bmatrix}
+
+        \\
+
+        R(\phi, \theta, \psi) = (R(\psi)R(\theta)R(\phi))^\intercal
+
+    """
+
+    if len(center) != 3:
+        raise ValueError(f'len(center) should be 3 but got {len(center)}')
+    center = np.array(center)
+
+    if len(axis_lengths) != 3:
+        raise ValueError(
+            f'len(axis_lengths) should be 3 but got {len(axis_lengths)}')
+    axis_lengths = np.array(axis_lengths)
+
+    if angles is None:
```

Comment:
Btw, you don't have to use rotation matrices. You can just define the `Rotation` object directly and use `Rotation.apply()` which is a lot more efficient.